demos/python_demos/formula_recognition_demo/utils.py

```diff
@@ -1,10 +1,341 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
 import json
+import logging as log
+import os
+import re
+import subprocess
+import tempfile
+from enum import Enum
+from multiprocessing.pool import ThreadPool
+
+import cv2 as cv
+import numpy as np
+import sympy
+from openvino.inference_engine import IECore
+from tqdm import tqdm
 
 START_TOKEN = 0
 END_TOKEN = 2
+DENSITY = 300
+DEFAULT_WIDTH = 800
+MIN_HEIGHT = 30
+MAX_HEIGHT = 150
+MAX_WIDTH = 1200
+MIN_WIDTH = 260
+COLOR_WHITE = (255, 255, 255)
+COLOR_BLACK = (0, 0, 0)
+RED = (0, 0, 255)
+# default value to resize input window's width in pixels
+DEFAULT_RESIZE_STEP = 10
+
+
+def strip_internal_spaces(text):
+    """
+    Removes spaces between digits, digit and dot,
+    dot and digit; after opening brackets and parentheses
+    and before closing ones; spaces around ^ symbol.
+    """
+    text = text.replace("{ ", "{")
+    text = text.replace(" }", "}")
+    text = text.replace("( ", "(")
+    text = text.replace(" )", ")")
+    text = text.replace(" ^ ", "^")
+    prev_text = ""
+    while prev_text != text:
+        prev_text = text
+        text = re.sub(r"([\d.]) ([\d.])", r"\1\2", text)
```

Comment:
Couldn't you do something like `text = re.sub(r'(?<=[\d.]) (?=[\d.])', '', text)` instead of looping?

Reply:
Updated regexp, thanks